### ISSUE TITLE
Syntax error, unexpected '\' (T_NS_SEPARATOR) when using Zend-Code 3.0.3

### ISSUE THREAD

**Issue by holtkamp** [Ocramius/ProxyManager#321]
Running a ```composer update``` resulted in:

```
  - Removing zendframework/zend-code (3.0.2)
  - Installing zendframework/zend-code (3.0.3)
    Downloading: 100%         
    Extracting archive

    REASON: ocramius/proxy-manager 2.0.2 requires zendframework/zend-code ~3.0 -> satisfiable by zendframework/zend-code[3.0.2, 3.0.0, 3.0.1, 3.0.3, 3.1.x-dev, 3.0.x-dev].
```

After that the following parse error occurs:

```
   [exec] Parse error: syntax error, unexpected '\' (T_NS_SEPARATOR), expecting identifier (T_STRING) in /workspace/project/vendor/ocramius/proxy-manager/src/ProxyManager/GeneratorStrategy/EvaluatingGeneratorStrategy.php(68) : eval()'d code on line 3
```

Moving back to ```"zendframework/zend-code" : "3.0.2"``` resolves the problem. 

The ProxyManager is used by www.php-di.org to define [lazy injected definitions](http://php-di.org/doc/lazy-injection.html) for the Dependency Injection Container.

Switched back to zend-code ```3.0.3``` and had the proxies written to file to check what generated code caused this:
```php
class Generated172c7b213948b8ab21f1c1c819c2a322 extends \\Doctrine\DBAL\Configuration implements \\ProxyManager\Proxy\VirtualProxyInterface
{

  ```
Note the double prefixing path separator ```\\```

 Might be a zend-code thingie, but as the Proxymanager complains, reported it here.

Probably caused by https://github.com/zendframework/zend-code/commit/d1f11ad95f9841f4553976eda1ec38ab68d53d39

**Comment by kdambekalns** [comment]
Can confirm the issue, using Neos and

- ocramius/proxy-manager 2.0.2
- zendframework/zend-code 3.0.3

as well.

**Comment by kdambekalns** [comment]
It's zend-code, see https://github.com/zendframework/zend-code/issues/74

**Comment by Ocramius** [comment]
Keeping it open, but marking it as invalid, since zendframework/zend-code#74 is the cause of this.

**Comment by Ocramius** [comment]
@holtkamp @kdambekalns can you try running a `composer update` and see if it actually works?

**Comment by holtkamp** [comment]
Yes, with zend-code "3.0.4" it works. Maybe you need to change the requirement in ```composer.json``` to specifically exclude 3.0.3 as suggested here: https://github.com/zendframework/zend-code/issues/74#issuecomment-229052073

**Comment by kdambekalns** [comment]
Late to the party, but yes: fixed with 3.0.3. Thanks!
